test/reserve/GenericDistributionHelper.e2e.test.ts

```diff
@@ -0,0 +1,360 @@
+      gdIn: ethers.utils.formatEther(amountToSell),
+      cusdOut: ethers.utils.formatUnits(quoteAmount, 6), // CUSD has 6 decimals
+      wxdcOut: ethers.utils.formatEther(quoteGasAmount)
+    });
+
+    // Set fee settings to trigger swap
+    const feeSettings = {
+      maxFee: ethers.utils.parseEther("100"),
+      minBalanceForFees: ethers.utils.parseEther("1"),
+      percentageToSellForFee: 5,
+      maxSlippage: 5
+    };
+
+    await deployer.sendTransaction({
+      to: XDC_ADDRESSES.Avatar,
+      value: ethers.utils.parseEther("1")
+    });
+
+    // Impersonate guardian to set fee settings
+    const avatarSigner = await ethers.getImpersonatedSigner(XDC_ADDRESSES.Avatar);
+    await distHelper.connect(avatarSigner).setFeeSettings(feeSettings);
+
+    // Ensure distHelper has low xdc balance to trigger swap
+    const currentxdcBalance = await ethers.provider.getBalance(distHelper.address);
+    console.log("Current xdc balance:", ethers.utils.formatEther(currentxdcBalance));
+    console.log("Min balance for fees:", ethers.utils.formatEther(feeSettings.minBalanceForFees));
+    if (currentxdcBalance.gte(feeSettings.minBalanceForFees)) {
+      // Send xdc token away to trigger swap
+      const tempAccount = ethers.Wallet.createRandom().connect(ethers.provider);
+      await deployer.sendTransaction({
+        to: tempAccount.address,
+        value: ethers.utils.parseEther("0.01")
+      });
+
+      // Transfer xdc balance from distHelper
+      const distHelperSigner = await ethers.getImpersonatedSigner(distHelper.address);
+
+      await distHelperSigner.sendTransaction({
+        to: tempAccount.address,
+        value: currentxdcBalance.sub(ethers.utils.parseEther("0.05"))
+      });
+    }
+
+    // Trigger distribution which should perform the swap
+    const xdcBalanceBefore = await ethers.provider.getBalance(distHelper.address);
+    const goodDollarBalanceBefore = await goodDollar.balanceOf(distHelper.address);
+
+    console.log("Balances before swap:", {
+      xdc: ethers.utils.formatEther(xdcBalanceBefore),
+      goodDollar: ethers.utils.formatEther(goodDollarBalanceBefore)
+    });
+
+    // Call onDistribution to trigger swap
+    await distHelper.onDistribution(0);
+
+    // Check balances after swap
+    const xdcBalanceAfter = await ethers.provider.getBalance(distHelper.address);
+    const goodDollarBalanceAfter = await goodDollar.balanceOf(distHelper.address);
+    console.log("Balances after swap:", {
+      xdc: ethers.utils.formatEther(xdcBalanceAfter),
+      goodDollar: ethers.utils.formatEther(goodDollarBalanceAfter)
+    });
+
+    // Verify swap occurred
+    // Either WXDC balance increased or xdc balance increased (after unwrapping)
+    const xdcIncrease = xdcBalanceAfter.sub(xdcBalanceBefore);
+
+    expect(xdcIncrease.gt(0), "Swap should have increased either WXDC or xdc balance").to.be.true;
+  });
+
+  it("should correctly calculate swap amounts using XSWAP pools", async function () {
```

Comment:
you are not comparing the quotes to calcgdtosell so you are not actually testing "correctly calculate swap amounts"